src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -407,7 +407,7 @@ export const DiscoverGrid = ({
 
   const { gridStyle, gridDensityClass } = useGridStyles({ rowDensityState, onUpdateDensity });
 
-  if (!rowCount && isLoading) {
```

Comment:
The problem with this approach is, it would load the grid every time data is fetched, which would lead to unwanted behavior when auto refresh is `on`. So it should be solved, when switching saves searches, like it's done when changing the index pattern, the global loading state should be displayed:

https://user-images.githubusercontent.com/463851/151855356-36190ff0-9bee-44f3-9f00-c8994687bfc5.mp4

Reply:
Think something like this would be the better solution for this:
`use_discover_state.ts` adding at L102 
```
/**
   * Reset to display loadinng spinner when savedSearch is changing
   */
    useEffect(()=> {
     reset();
   },[savedSearch.id, reset]);
**/
So every time the savedSearch.id changes, the loading spinner is displayed ... this would be the same behavior like switching the data view, WDYT?

Reply:
It should work, agree